code/cycler/src/wattrex_battery_cycler/app/app_man/app_man_core.py

```diff
@@ -232,22 +244,30 @@ def execute_machine_status(self) -> None: #pylint: disable=too-many-branches
             ## Process machine status
             if self.state == AppManCoreStatusE.GET_EXP:
                 ## Wait for cs status to continue
-                if ((not self.__wait_cs_reqst and not self.__wait_exp_reqst) or
-                    self.__iter> _PERIOD_WAIT_EXP):
+                if self.__get_exp_status is _AppManCoreGetExpStatusE.GET_EXP:
                     self.__fetch_new_exp()
                     self.__request_cs_status()
-                ## Check if the cycler station is deprecated
-                if self.is_deprecated:
-                    log.critical("Cycler station is deprecated")
-                    self.turn_deprecated()
-                    self.state = AppManCoreStatusE.ERROR
-                ## Wait for the experiment and check if the experiment is valid
-                if not self.experiment is None and self.__wait_exp_reqst:
-                    self.__iter = 0
-                    self.__wait_exp_reqst = False
-                    self.state = AppManCoreStatusE.PREPARE_EXP
-                elif self.__wait_exp_reqst:
-                    self.__iter +=1
+                    self.__get_exp_status = _AppManCoreGetExpStatusE.WAIT_CS
+                elif self.__get_exp_status is _AppManCoreGetExpStatusE.WAIT_CS:
+                    if not self.__wait_cs_reqst:
+                        if self.is_deprecated:
+                            log.critical("Cycler station is deprecated")
+                            self.turn_deprecated()
+                            self.state = AppManCoreStatusE.ERROR
+                        else:
+                            self.__get_exp_status = _AppManCoreGetExpStatusE.WAIT_EXP
+                elif self.__get_exp_status is _AppManCoreGetExpStatusE.WAIT_EXP:
+                    if self.experiment is not None and self.__wait_exp_reqst:
```

Comment:
I think it should be: `... and not self.__wait_exp_reqst`

Reply:
Probably, changing it